spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java

```diff
@@ -1039,12 +1039,16 @@ protected File get(Message<?> message, Session<F> session, String remoteDir, Str
 			if (!appending && !tempFile.renameTo(localFile)) {
 				throw new MessagingException("Failed to rename local file");
 			}
-			if (this.options.contains(Option.PRESERVE_TIMESTAMP)) {
+			if (this.options.contains(Option.PRESERVE_TIMESTAMP)
+					|| FileExistsMode.REPLACE_IF_MODIFIED.equals(fileExistsMode)) {
 				localFile.setLastModified(getModified(fileInfo));
 			}
 		}
 		else if (FileExistsMode.REPLACE_IF_MODIFIED.equals(fileExistsMode)) {
 			logger.debug("Local file '" + localFile + "' has the same modified timestamp, ignored");
```

Comment:
I don't know what you mean by "move to the else side". The MGET logic has no bearing; it's just a signal to the caller to not include the file in the output message. The log should be emitted, regardless.

Right; no backport.